doc/source/user_guide/indexing.rst

```diff
@@ -1158,6 +1158,40 @@ Mask
    s.mask(s >= 0)
    df.mask(df >= 0)
 
+.. _indexing.np_where:
+
+Setting with enlargement conditionally using :func:`numpy`
+----------------------------------------------------------
+
+An alternative to :meth:`~pandas.DataFrame.where` is to use :func:`numpy.where`.
```

Comment:
What is the difference compared to using `DataFrame.where` ?

Reply:
Hi Joris, with `np.where`, enlarging the dataframe by adding a column is straightforward.  Also, it naturally extends to multiple conditions with `np.select`.

Reply:
@suvayu thanks for the answer! Looking at the example now, I think the main difference is that with np.where, your x/y values to take from don't need to be a Series to start with (eg `Series.where` always takes the values from the calling Series where the condition is True, and so indeed you wouldn't be able to achieve the actual example here)